### PR TITLE
Pull for review request

### PR DESCRIPTION
**Pull request by jjdelvalle**
All code is part of this PR. It was created for review purposes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Language Technologies in Practice Final Project - Sarcasm Detection
```
Comment (jirka-x1) [inline README.md:1]: You should provide an example that would demonstrate what your code does - I clone the repository, what should I do next to quickly see its abilities.

File: main.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        df["text"] = df["text"].str.replace(r"@[a-z0-9_]+", r"_mention", regex=True)
+        df["text"] = df["text"].str.replace(r"https?://[^\s]+", "_url ", regex=True)
+
+        # Separate out symbols since tokenizers might not handle them properly.
+        # as per fastText documentation:
+        df["text"] = df["text"].str.replace(r"([\.\\!\?,'/\(\)])", r" \1 ", regex=True)
+    return df
+
+def prepare_data(input_path: str):
+    train = read_data(f"{input_path}/train")
```
Comment (jirka-x1) [inline main.py]: this is fine and don't change it. But have a look at pathlib. There you can write:\

def prepare_data(input_path: Path):
    train = read_data(input_path / "train")

File: main.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        # for the fasttext tokenizer
+        fast_df.to_csv(out_name, sep='\t', header=False, index=False)
+
+def main(args):
+    # Spliting into X & y
+    X_train, y_train, X_test, y_test = prepare_data(args.input_path)
+
+    # Building a TF IDF matrix out of the corpus of reviews
+    if args.vectorizer == 'tfidf':
+        from sklearn.feature_extraction.text import TfidfVectorizer
```
Comment (jirka-x1) [inline main.py]: move each of these into a separate function (that way it is easier to see what is the input and output, plus you can use self-explanatory function names, and comments)

File: README.md (hunk 1 above)
Comment (jjdelvalle) [reply README.md:1]: Added some examples to the readme file

File: main.py (hunk 2 above)
Comment (jjdelvalle) [reply main.py]: Decided to adopt using `Path`, I knew about it but was too lazy to use it from the get go hah.

File: main.py (hunk 3 above)
Comment (jjdelvalle) [reply main.py]: Done.

**Comment by jjdelvalle** [comment]
Made some changes according to you feedback, @jirka-x1. Thanks for the feedback!

**Review by jirka-x1 — APPROVED** [review]

**Pull request merged**